src/itsdangerous/serializer.py

```diff
@@ -71,7 +78,7 @@ class Serializer:
     #: The default serialization module to use to serialize data to a
     #: string internally. The default is :mod:`json`, but can be changed
     #: to any object that provides ``dumps`` and ``loads`` methods.
-    default_serializer: t.Any = json
+    default_serializer: _PDataSerializer[_TAnyStr] = json  # type: ignore[assignment]
```

Comment:
```suggestion
    default_serializer: _PDataSerializer[str] = json
```
This seems more sane to me, if you're worried about subclasses providing a `bytes` serializer as default, you could change it to `_PDataSerializer[t.Any]` or a very long-winded `t.Union[_PDataSerializer[str], _PDataSerializer[bytes]]`. As a class attribute it shouldn't really be generic, that's prone to weird errors.